### PR TITLE
Add `lib` chart for shared templates between `seqr`, `hail-search`, and `pipeline-runner` charts.

### PR DESCRIPTION
**Pull request by bpblanken**
- Moves hail search volume to a shared location where it will be used by the other charts.
- Enables support for a `local` persistent volume, as opposed to `csi` (used by us).
- The shared volume will be mounted to `/seqr`, and all seqr application state will persist there... at this point that will include reference data hail tables, vep reference data, luigi scheduler state, the postgres `PGDATA` dir, and the hail search data itself.  (this is maybe a kubernetes anti-pattern but is very simple and preserves what we were doing with the `./data` directory in the old `docker-compose`).

### CODE REVIEW

File: charts/lib/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+apiVersion: v2
+name: lib
+description: A collection of shared resources used by the comprising helm charts of Seqr, an open source software platform for rare disease genomics
```
Comment (hanars) [inline charts/lib/Chart.yaml:3]: comprising?

File: charts/lib/values.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+exports:
+  global:
+    lib:
+      persistentVolume:
+        storageCapacity: '750Gi'
+        accessMode: ReadWriteOnce
```
Comment (hanars) [inline charts/lib/values.yaml:6]: Is this the correct access mode for local installs? If this is where all data goes including postgres I do not think that will work

Comment (bpblanken) [reply charts/lib/values.yaml:6]: Yes... the `ReadWriteOnce` means it's loaded on one `node` but can be loaded read-write by multiple pods.

Comment (bpblanken) [reply charts/lib/values.yaml:6]: For more context, the alternative `ReadWriteMany` is, I believe, not supported for `local` volumes and is more meant for something like `NFS`.

**Review by hanars — APPROVED** [review]

**Review by sjahl — APPROVED** [review]
This looks good to me! The only thing I'd be concerned about with the shared /seqr mount is pods theoretically being able to access data that they shouldn't (e.g. a not-database pod accessing the PGDATA dir). But, if we're trying to keep things simple and compatible with the old deployment pattern, not a huge issue.

**Comment by bpblanken** [comment]
```
not-database pod accessing the PGDATA dir
```

yes this is definitely possible!  I will think a little more about this... but yes, simplicity was the goal here.

**Pull request merged**